### PR TITLE
Left-align tooltips for :before decorators

### LINKED ISSUE #159225: Tooltips for :before decorators should be aligned with content
Does this issue occur when all extensions are disabled?: No -- requires any extension that displays :before decorator content.

- VS Code Version: 1.70.2
- OS Version: macOS Monterey 12.3.1

Our extension creates a decorator that adds a gutter icon to the left of the editor text, by setting `ThemableDecorationAttachmentRenderOptions.contentIconPath` on `ThemableDecorationInstanceRenderOptions.before`.  A tooltip is also provided.  The goal is to provide contextual information on source code, with a brief explanation in the tooltip, plus a link to where more information can be displayed.  Here is an image of a demo extension showing something like this:

Screen Shot 2022-08-25 at 1 07 32 PM

(I have uploaded a demo of the extension code for the above image here, in case it is useful: https://github.com/matthewjamesadam/gutter-icon-extension)

The natural workflow is that a user would hover over the icon, which causes the tooltip to display.  Then they would move their mouse to the title, in order to click on it to get more information, as in the green arrow here:
Screen Shot 2022-08-25 at 1 08 41 PM

This workflow does not work as expected however, because only the areas shaded green in the image below will allow the tooltip to continue displaying.  If the mouse moves into the red zone (or anywhere outside of green), the tooltip disappears:

Screen Shot 2022-08-25 at 1 08 41 PM copy

This makes the tooltip almost impossible to use in practice, as moving the mouse towards the title causes the tooltip to disappear:

CleanShot 2022-08-25 at 13 19 08

This is heavily impacting the usability of our extension, as users are finding the gutter icon does not work as they would expect.

I believe the desired behaviour for these tooltips is to be left-aligned with the `before` content, not with the left edge of the editor text.  This is both consistent with other tooltips, and makes the tooltip easier to navigate to from the `before` content.

Screen Shot 2022-08-25 at 1 22 37 PM

I have prepared a PR that fixes this issue and will refer to this issue from it shortly.  Thanks!

### PR DESCRIPTION
**Pull request by matthewjamesadam**
Fixes https://github.com/microsoft/vscode/issues/159225

Left-align tooltip content for `:before` decorators.  This fixes issues where tooltips are very difficult to click on from decorator gutter icons.

I admit that my solution here is likely not ideal -- I am not 100% familiar with every part of this and made some guesses.  I'm open to any alternative approach to resolving this problem.

Thanks!

### CODE REVIEW

**Comment by Maungmaunglwin930** [comment]
I played the game and can't withdraw money from the game I don't know what's needed, I'm not good at technology, it's just trying to get the money in the game automatically with one click, so I'm just trying it out, if I make a mistake, please forgive me.

File: src/vs/editor/browser/widget/codeEditorWidget.ts [hunk 1]
```diff
@@ -1360,6 +1360,7 @@ export class CodeEditorWidget extends Disposable implements editorBrowser.ICodeE
 			if (decorationOption.hoverMessage) {
 				opts.hoverMessage = decorationOption.hoverMessage;
 			}
+
 			newModelDecorations.push({ range: decorationOption.range, options: opts });
```
Comment (hediet) [inline src/vs/editor/browser/widget/codeEditorWidget.ts:1364]: Nit: Please don't touch unrelated files.

Comment (matthewjamesadam) [reply src/vs/editor/browser/widget/codeEditorWidget.ts]: Thanks, I've removed this.

**Review by Maungmaunglwin930 — APPROVED** [review]
Please help me

**Comment by Maungmaunglwin930** [comment]
Please help me
